hazelcast/src/main/java/com/hazelcast/jet/impl/SnapshotValidator.java

```diff
@@ -47,8 +47,8 @@ static long validateSnapshot(
         if (validationRecord == null) {
             String nameOrId = snapshotName != null ? '"' + snapshotName + '"' : "with ID " + snapshotId;
             throw new JetException(String.format(
-                    "snapshot %s doesn't exist or is damaged. Unable to restore the state for %s.",
-                    nameOrId, jobIdString));
+                    "snapshot %s in IMap %s (%d entries) is damaged. Unable to restore the state for %s.",
+                    nameOrId, snapshotMap.getName(), snapshotMap.size() - 1, jobIdString));
```

Comment:
When doing `snapshotMap.size() - 1`, I think what we removed was actually the validation record entry below, it's better to not subtract 1 in this case since that validation record isn't here already. I overlooked this in the first original pr.

Reply:
Yes, you are right. Fixed.